### PR TITLE
Feature/enemy card tracker

### PR DESCRIPTION
**Pull request by austinbrianrogers**
LAST COMMIT EVAR.

### CODE REVIEW

File: UI/CommanderContext.cpp [hunk 1]
```diff
@@ -233,4 +240,11 @@ namespace userinterface
 		nextTurnButton->getTransform().place(winX - 160, winY - bgScale.y - 5, -0.01f);
 		nextTurnButton->getTransform().scale2D(160, 105);
 	}
+
+	void CommanderContext::receiveCommanderDamage(kitten::Event::EventType p_type, kitten::Event* p_event)
+	{
+		int dmg = p_event->getInt(DAMAGE_AS_INT);
+		m_currentHP += dmg;
```
Comment (calmackenzie) [inline UI/CommanderContext.cpp]: += ?

File: components/DeckCountTextureBind.cpp [hunk 2]
```diff
@@ -38,18 +49,49 @@ void DeckCountTextureBind::start()
...
 
-	kitten::K_GameObject* counter = kitten::K_GameObjectManager::getInstance()->createNewGameObject("ui/deck/deck_counter_textbox.json");
-	puppy::TextBox* txtBoxComp = counter->getComponent<puppy::TextBox>();
-	txtBoxComp->setText("Loading...");
 	const glm::vec2 deckScale = getTransform().getScale2D();
 	const glm::vec3 deckTrans = getTransform().getTranslation();
-	counter->getTransform().place2D(deckTrans.x + 20, deckTrans.y + deckScale.y + 10);
+
+	bool host = false;
+	if (networking::ClientGame::getClientId == 0)
```
Comment (calmackenzie) [inline components/DeckCountTextureBind.cpp]: could have just put `bool host = (networking::ClientGame::getClientId == 0);` :p

File: UI/CommanderContext.cpp [hunk 3]
```diff
@@ -53,6 +53,11 @@ namespace userinterface
 			attachCommander(kibble::getUnitFromId(29));
 			//REMOVE WHEN NOT TESTING
 		}
+
+		kitten::EventManager::getInstance()->addListener(
+			kitten::Event::EventType::Damage_Commander,
+			this,
+			std::bind(&CommanderContext::receiveCommanderDamage, this, std::placeholders::_1, std::placeholders::_2));
```
Comment (calmackenzie) [inline UI/CommanderContext.cpp]: When does this deregister?

File: components/DeckCountTextureBind.cpp (hunk 2 above)
Comment (austinbrianrogers) [reply components/DeckCountTextureBind.cpp]: It isn't. Fixing.

**Pull request merged**